modules/xcm-bridge-hub/src/lib.rs

```diff
@@ -0,0 +1,970 @@
+		// `T as pallet_bridge_messages::Config<T::BridgeMessagesPalletInstance>::BridgedChain::NetworkId`
+		/// Bridged network id.
+		#[pallet::constant]
+		type BridgedNetworkId: Get<NetworkId>;
+		/// Associated messages pallet instance that bridges us with the
+		/// `BridgedNetworkId` consensus.
+		type BridgeMessagesPalletInstance: 'static;
+
+		/// A set of XCM locations within local consensus system that are allowed to open
+		/// bridges with remote destinations.
+		// TODO: there's only one impl of `EnsureOrigin<Success = MultiLocation>` -
+		// `EnsureXcmOrigin`, but it doesn't do what we need. Is there some other way to check
+		// `Origin` and get matching `MultiLocation`???
+		type AllowedOpenBridgeOrigin: EnsureOrigin<
+			<Self as SystemConfig>::RuntimeOrigin,
+			Success = MultiLocation,
+		>;
+		/// A converter between a multi-location and a sovereign account.
+		type BridgeOriginAccountIdConverter: ConvertLocation<Self::AccountId>;
+
+		/// Amount of this chain native tokens that is reserved on the sibling parachain account
+		/// when bridge open request is registered.
+		#[pallet::constant]
+		type BridgeReserve: Get<BalanceOf<ThisChainOf<Self, I>>>;
+		/// Currency used to pay for bridge registration.
+		type NativeCurrency: ReservableCurrency<Self::AccountId>;
+	}
+
+	/// An alias for the bridge metadata.
+	pub type BridgeOf<T, I> = Bridge<ThisChainOf<T, I>>;
+	/// An alias for the this chain.
+	pub type ThisChainOf<T, I> =
+		pallet_bridge_messages::ThisChainOf<T, <T as Config<I>>::BridgeMessagesPalletInstance>;
+	/// An alias for the associated lanes manager.
+	pub type LanesManagerOf<T, I> =
+		pallet_bridge_messages::LanesManager<T, <T as Config<I>>::BridgeMessagesPalletInstance>;
+
+	#[pallet::pallet]
+	pub struct Pallet<T, I = ()>(PhantomData<(T, I)>);
+
+	#[pallet::call]
+	impl<T: Config<I>, I: 'static> Pallet<T, I>
+	where
+		T: frame_system::Config<
+			AccountId = AccountIdOf<ThisChainOf<T, I>>,
+			BlockNumber = BlockNumberOf<ThisChainOf<T, I>>,
+		>,
+		T::NativeCurrency: Currency<T::AccountId, Balance = BalanceOf<ThisChainOf<T, I>>>,
+	{
+		/// Open a bridge between two locations.
+		///
+		/// The caller must be within the `T::AllowedOpenBridgeOrigin` filter (presumably: a sibling
+		/// parachain or a parent relay chain). The `bridge_destination_relative_location` must be a
+		/// destination within the consensus of the `T::BridgedNetworkId` network.
+		///
+		/// The `BridgeReserve` amount is reserved on the caller account. This reserve
+		/// is unreserved after bridge is closed.
+		///
+		/// The states after this call: bridge is `Opened`, outbound lane is `Opened`, inbound lane
+		/// is `Opened`.
+		#[pallet::call_index(0)]
+		#[pallet::weight(Weight::zero())] // TODO: https://github.com/paritytech/parity-bridges-common/issues/1760 - weights
+		pub fn open_bridge(
+			origin: OriginFor<T>,
+			bridge_destination_relative_location: Box<VersionedMultiLocation>,
```

Comment:
I've been looking at `pallet-xcm` ' `send_message`, but you're right - this method will be called by sibling/parent chains and it'd be better to avoid strange and confusing conversions of this argument while it heads to this runtime. Done